aws/templates/solution/solution_s3.ftl

```diff
@@ -57,38 +53,46 @@
             [/#if]
         [/#list]
 
-        [#if solution.PublicAccess.Enabled ]
-
-            [#assign bucketPolicyId = resources["bucketpolicy"].Id ]
-
-            [#assign policyStatements = [] ]
-
-            [#switch solution.PublicAccess.Permissions ]
-                [#case "ro" ]
-                    [#assign policyStatements = s3ReadPermission(
-                                                    s3Name,
-                                                    publicPrefix,
-                                                    "*",
-                                                    "*",
-                                                    publicIPWhiteList)]
-                    [#break]
-                [#case "wo" ]
-                    [#assign policyStatements = s3WritePermission(
-                                                    s3Name,
-                                                    publicPrefix,
-                                                    "*",
-                                                    "*",
-                                                    publicIPWhiteList)]
-                    [#break]
-                [#case "rw" ]
-                    [#assign policyStatements = s3AllPermission(
-                                                    s3Name,
-                                                    publicPrefix,
-                                                    "*",
-                                                    "*",
-                                                    publicIPWhiteList)]
-                    [#break]
-            [/#switch]
+        [#assign bucketPolicyId = resources["bucketpolicy"].Id ]
+        [#assign policyStatements = [] ]
+
+        [#list solution.PublicAccess?values as publicAccessConfiguration]
+            [#list asArray(publicAccessConfiguration.Paths) as publicPrefix]
```

Comment:
@ml019 I used `asArray` function to convert a single string value into an array, is it a correct approach?

Reply:
Also, this code produces a statement per path, but we could change it to a single statement with a Resource as an array instead of a single string. Shall we do it? It will require the changes in getS3Statement and getPolicyStatement functions.

Reply:
Because of the type in the config, Paths will already be an array so no need to do asArray(). It won't hurt but is redundant.

https://github.com/codeontap/gen3/blob/master/aws/templates/common.ftl#L586
https://github.com/codeontap/gen3/blob/master/aws/templates/base.ftl#L328

For now I think a single statement per path is simpler/cleaner. If we find the need for a very long public access list then refactoring may be in order. I suspect there will typically not be a very large list.